core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/machineLearning/Scorer/ScorerOpDesc.scala

```diff
@@ -0,0 +1,189 @@
+      new JsonSchemaString(path = HideAnnotation.hideType, value = HideAnnotation.Type.equals),
+      new JsonSchemaString(path = HideAnnotation.hideExpectedValue, value = "false")
+    )
+  )
+  var regressionMetrics: List[regressionMetricsFnc] = List()
+
+  override def operatorInfo: OperatorInfo =
+    OperatorInfo(
+      "Scorer",
+      "Scorer for machine learning models",
+      OperatorGroupConstants.MODEL_PERFORMANCE_GROUP,
+      inputPorts = List(InputPort()),
+      outputPorts = List(OutputPort())
+    )
+  override def getOutputSchema(schemas: Array[Schema]): Schema = {
+    val outputSchemaBuilder = Schema.builder()
+    outputSchemaBuilder.add(new Attribute("Label", AttributeType.STRING))
+
+    if (isRegression) {
+      regressionMetrics.foreach(metric => {
+        outputSchemaBuilder.add(new Attribute(metric.getName(), AttributeType.DOUBLE))
+      })
+    } else {
+      classificationMetrics.foreach(metric => {
+        outputSchemaBuilder.add(new Attribute(metric.getName(), AttributeType.DOUBLE))
+      })
+    }
+
+    outputSchemaBuilder.build()
+  }
+
+  private def getClassificationScorerName(scorer: classificationMetricsFnc): String = {
+    // Directly return the name of the scorer using the getName() method
+    scorer.getName()
+  }
+  private def getRegressionScorerName(scorer: regressionMetricsFnc): String = {
+    // Directly return the name of the scorer using the getName() method
+    scorer.getName()
+  }
+
+  private def getEachScorerName(scorer: Any): String =
+    scorer match {
+      case s: classificationMetricsFnc => getClassificationScorerName(s)
+      case s: regressionMetricsFnc     => getRegressionScorerName(s)
+      case _                           => throw new IllegalArgumentException("Unknown scorer type")
+    }
+
+  private def getSelectedScorers(): String = {
+    // Return a string of scorers using the getEachScorerName() method
+    var scorers: List[_] = List()
+    if (isRegression) scorers = regressionMetrics
+    else scorers = classificationMetrics
+
+    scorers.map(scorer => getEachScorerName(scorer)).mkString("'", "','", "'")
+  }
+
+  override def generatePythonCode(): String = {
+    var is_regression = "False"
+    if (isRegression) is_regression = "True"
+    val finalcode =
+      s"""
+         |from pytexera import *
+         |import pandas as pd
+         |import numpy as np
+         |from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, mean_squared_error, root_mean_squared_error, mean_absolute_error, r2_score
+         |import json
+         |
+         |
+         |def classification_scorers(y_true, y_pred, scorer_list, labels):
+         |  result = {scorer: [None] * len(labels) for scorer in scorer_list}
+         |  metrics_func = {'Precision Score': precision_score, 'Recall Score': recall_score, 'F1 Score': f1_score}
+         |
+         |  for scorer in scorer_list:
+         |    prediction = None
```

Comment:
your operator is named `scorer`. don't name each metric as `scorer`.

Reply:
fix